provision/service.go

```diff
@@ -192,11 +207,49 @@ func (ps *provisionService) Provision(token, externalID, externalKey string) (re
 			}
 			res.Whitelisted[thing.ID] = true
 		}
+
 	}
 
+	ps.updateGateway(token, bs, channels)
 	return res, nil
 }
 
+func (ps *provisionService) updateGateway(token string, bs SDK.BootstrapConfig, channels []SDK.Channel) error {
+	g := Gateway{}
+	for _, ch := range channels {
+		switch ch.Metadata["type"] {
+		case control:
+			g.CtrlChannelID = ch.ID
+		case data:
+			g.DataChannelID = ch.ID
+		case export:
+			g.ExportChannelID = ch.ID
+		}
+	}
+	g.GwPassword = bs.ExternalKey
+	g.MAC = bs.ExternalID
+	// same value, keep external id in meta for compatibility
```

Comment:
I did not quite understand this comment. Also, they start with capital letters.

Reply:
this  `MAC` property will be removed, actually

Reply:
Why not removing it now then?